siatest/testgroup.go

```diff
@@ -244,25 +251,32 @@ func fundNodes(miner *TestNode, nodes map[*TestNode]struct{}) error {
 	return nil
 }
 
-// hostsInRenterDBCheck makes sure that all the renters see numHosts hosts in
-// their database.
-func hostsInRenterDBCheck(miner *TestNode, renters map[*TestNode]struct{}, numHosts int) error {
+// hostsInRenterDBCheck makes sure that all the renters see all hosts in their
+// database.
+func hostsInRenterDBCheck(miner *TestNode, renters map[*TestNode]struct{}, hosts map[*TestNode]struct{}) error {
 	for renter := range renters {
-		err := Retry(100, 100*time.Millisecond, func() error {
-			hdag, err := renter.HostDbActiveGet()
-			if err != nil {
-				return err
-			}
-			if len(hdag.Hosts) != numHosts {
-				if err := miner.MineBlock(); err != nil {
-					return err
+		for host := range hosts {
+			numRetries := 0
+			err := Retry(100, 100*time.Millisecond, func() error {
+				numRetries++
+				if renter == host {
+					// We don't care if the renter is also a host.
+					return nil
 				}
-				return errors.New("renter doesn't have enough active hosts in hostdb")
+				// Check if the renter has the host in its db.
+				if err := renter.knowsHost(host); err != nil {
+					if numRetries%10 == 0 {
+						if err := miner.MineBlock(); err != nil {
+							return err
```

Comment:
This is too much indentation, can you re-structure this function to clean that up a bit?